hydromt/cli/main.py

```diff
@@ -464,31 +465,58 @@ def export(
     if dd and "deltares_data" not in data_libs:  # deltares_data from cli
         data_libs = ["deltares_data"] + data_libs  # prepend!
 
-    if export_config:
-        args = export_config.pop("args", {})
+    sources: List[str] = []
+
+    if source:
+        if isinstance(source, str):
+            sources = [source]
+        else:
+            sources = list(source)
+
+    if config:
+        config_dict = cli_utils.parse_config(config)["export_data"]
+        args = config_dict.pop("args", {})
         if "catalog" in args.keys():
             data_libs = data_libs + args.pop("catalog")
         time_tuple = args.pop("time_tuple", None)
         region = region or args.pop("region", None)
         if isinstance(region, str):
             region = json_decode(region)
+
+        sources = sources + config_dict["sources"]
+
+    data_catalog = DataCatalog(data_libs=data_libs)
+    _ = data_catalog.sources  # initialise lazy loading
+
+    if region:
+        if "bbox" in region:
+            bbox = region["bbox"]
+        elif "geom" in region:
+            bbox = GeoDataFrame.from_file(region["geom"]).total_bounds
+        else:
+            raise NotImplementedError("Only bbox and geom are supported for export")
+
+        if not set(region.keys()).issubset({"bbox", "geom"}):
+            logger.warning(
+                "Found unsupported arguments for region in addition to bbox or geom. these will be ignored"
+            )
     else:
-        time_tuple = None
-        region = None
+        bbox = None
 
-    if target:
-        export_targets = [{"source": target}]
-    elif export_config:
-        export_targets = export_config["sources"]
+    if time_tuple:
+        tup = literal_eval(time_tuple)
+        time_start = datetime.strptime(tup[0], "%Y-%m-%d")
+        time_end = datetime.strptime(tup[1], "%Y-%m-%d")
+        time_tup = (time_start, time_end)
     else:
-        export_targets = None
+        time_tup = None
 
     try:
-        data_catalog = DataCatalog(data_libs=data_libs)
         data_catalog.export_data(
             export_dest_path,
-            source_names=export_targets,
-            time_tuple=time_tuple,
+            source_names=sources,
+            bbox=bbox,
+            time_tuple=time_tup,
```

Comment:
I miss passing the other arguments from config_dict like unit_conversion etc. (especially if you pop the others, can easily added here as **config_dict)

Reply:
Oh sorry, what you did also works perfectly :) I meant pass here export_data(source_names = sources, ..., **config_dict) and instead of defining unit_conversion = True etc. just define config_dict = dict() in the case there is no config file provided.
But his also works perfectly well